features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/repository/DefaultFlagsRepository.kt

```diff
@@ -69,18 +80,31 @@ internal class DefaultFlagsRepository(
         return null
     }
 
-    override fun getEvaluationContext(): EvaluationContext? = atomicState.get()?.context
+    override fun getEvaluationContext(): EvaluationContext? {
+        waitForPersistenceLoad()
+        return atomicState.get()?.context
+    }
 
     @Suppress("ReturnCount")
     override fun getPrecomputedFlagWithContext(key: String): Pair<PrecomputedFlag, EvaluationContext>? {
+        waitForPersistenceLoad()
         val state = atomicState.get() ?: return null
         val flag = state.flags[key] ?: return null
         return flag to state.context
     }
 
+    private fun waitForPersistenceLoad() {
+        try {
+            persistenceLoadedLatch.await(persistenceLoadTimeoutMs, TimeUnit.MILLISECONDS)
+        } catch (e: InterruptedException) {
+            Thread.currentThread().interrupt()
```

Comment:
`InterruptedException` is thrown if the countdown latch does not open within the timeout. What is the effect of interrupting the current thread here instead of log and return?

Reply:
log and return would swallow the interrupt, but calling Thread.currentThread().interrupt() preserves it for callers further up the stack